lib/haml/railtie.rb

```diff
@@ -10,13 +10,32 @@
 module Haml
   class Railtie < ::Rails::Railtie
     initializer :haml do |app|
-      require "haml/template"
-      if defined?(::Sass::Rails::SassTemplate) && app.config.assets.enabled
-        require "haml/sass_rails_filter"
+      if defined?(ActiveSupport)
+        ActiveSupport.on_load(:before_initialize) do
+          ActiveSupport.on_load(:action_view) do
```

Comment:
I guess we can remove either `on_load(:before_initialize)` or `on_load(:action_view)`. Why are both of them required here?